src/gui/application.cpp

```diff
@@ -487,32 +486,32 @@ void Application::slotShowGuiMessage(const QString &title, const QString &messag
 }
 
 
-void Application::slotownCloudWizardDone(int res)
+AccountStatePtr Application::addNewAccount(AccountPtr newAccount)
 {
-    AccountManager *accountMan = AccountManager::instance();
-    FolderMan *folderMan = FolderMan::instance();
+    auto *accountMan = AccountManager::instance();
 
-    // During the wizard, scheduling of new syncs is disabled
-    folderMan->setSyncEnabled(true);
+    // first things first: we need to add the new account
+    auto accountStatePtr = accountMan->addAccount(newAccount);
 
-    if (res == QDialog::Accepted) {
-        // Check connectivity of the newly created account
-        _checkConnectionTimer.start();
-        slotCheckConnection();
+    // check connectivity of the newly created account
+    _checkConnectionTimer.start();
+    slotCheckConnection();
 
-        // If one account is configured: enable autostart
-        bool shouldSetAutoStart = (accountMan->accounts().size() == 1);
+    // if one account is configured: enable autostart
+    bool shouldSetAutoStart = (accountMan->accounts().size() == 1);
```

Comment:
I've checked how this is stored at the moment. We'd need to store whether to autostart or not in our config, not just autodetect it from the system. Then, we can also specify a default value on first run more easily (right now, we can't really tell whether this is a first run or not).

I'd do any modifications in a separate follow-up PR, as it will take some testing to make sure we get it right and it's a slightly complex change.